### PR TITLE
Improve premiere timestamp to be easier to read

### PR DESCRIPTION
**Pull request by Aiz0**
---
Improve premiere timestamp to be easier to read
---

**Important note**
We may remove your pull request if you do not use this provided PR template correctly.

**Pull Request Type**
Please select what type of pull request this is:
- [ ] Bugfix
- [X] Feature Implementation
- [ ] Documentation
- [ ] Other

**Related issue**
no related issue that i know of

**Description**
Changes the premiere timestamp to be easier to read.
Removes year and seconds from string because they are not important.
Displays month as it's "long name" (eg., March)

also displays timestamp in the same locale set in freetube.

**Screenshots (if appropriate)**
before:
my locale
2022-09-10_T00:28:02

after:
my locale / system default
2022-09-10_T00:10:35
english locale
2022-09-10_T00:10:55
french locale
2022-09-10_T01:11:38

**Testing (for code that is not small enough to be easily understandable)**
I have tested some different locales in freetube.
Open this in browser and you should be able to find some upcoming live streams.
https://www.youtube.com/live

**Desktop (please complete the following information):**
 - OS Version: [e.g. 22]
 - FreeTube version: [e.g. 0.8]

**Additional context**
~~I noticed that the locale set in freetube was not used when using `date.toLocaleString()`.
So I searched, found, and copied the code from `watch-video-info.js` to solve this problem.
This obviously however caused 'en' to be used when i would have preferred my system locale.
I changed 'en' to 'default' to solve that.
Should the same be done to the other functions that do this?~~

### CODE REVIEW

File: src/renderer/views/Watch/Watch.js [hunk 1]
```diff
@@ -163,6 +163,9 @@ export default Vue.extend({
     },
     theatrePossible: function () {
       return !this.hideRecommendedVideos || (!this.hideLiveChat && this.isLive) || this.watchingPlaylist
+    },
+    currentLocale: function () {
+      return this.$store.getters.getCurrentLocale
```
Comment (absidue) [inline src/renderer/views/Watch/Watch.js:168]: Please don't use `getCurrentLocale`, instead do it the same way that is done in https://github.com/FreeTubeApp/FreeTube/pull/2557.

Comment (Aiz0) [reply src/renderer/views/Watch/Watch.js]: This should be fixed now.
You were right that I didn't need to specify 'default' when doing it this way.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
this closes https://github.com/FreeTubeApp/FreeTube/issues/1429 right?

**Comment by Aiz0** [comment]
> this closes #1429 right?

I don't think so.
That issue has to do with the flatpak build using UTC instead of the users timezone.
Locale was the same for both, i posted an image in there comparing two builds and only the timezone differed in how the timestamp was displayed.

**Comment by Aiz0** [comment]
I found out about [Intl.RelativeTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) and am using it to simplify the logic used to get `upcomingTimeLeft`.
Should I add these changes to this pull request or would you rather I open a new one later?

File: src/renderer/views/Watch/Watch.js [hunk 2]
```diff
@@ -409,7 +413,13 @@ export default Vue.extend({
 
             if (typeof startTimestamp !== 'undefined') {
               const upcomingTimestamp = new Date(result.videoDetails.liveBroadcastDetails.startTimestamp)
-              this.upcomingTimestamp = upcomingTimestamp.toLocaleString()
+              const timestampOptions = {
+                month: 'long',
+                day: 'numeric',
+                hour: 'numeric',
+                minute: '2-digit'
```
Comment (ChunkyProgrammer) [inline src/renderer/views/Watch/Watch.js:420]: I think you should include year as well. Example: https://www.youtube.com/watch?v=O3cCYok_ukk (this video is to be published in 2024)

Comment (Aiz0) [reply src/renderer/views/Watch/Watch.js:420]: Thanks for the video.

I've added a conditional so year is displayed if the video isn't published in current year

**Review by ChunkyProgrammer — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by PikachuEXE — APPROVED** [review]

**Pull request merged**